### PR TITLE
[Testing] Use a better example for Symfony asserts

### LINKED ISSUE #13036: assertSelectorTextSame bad example
As discussed in https://github.com/symfony/symfony/issues/35191#event-2958352022,
@xabbuh recommended the [docs](https://symfony.com/doc/current/testing/functional_tests_assertions.html) could be improved by pointing out that the example does work because the html tag is unique.

I oversaw that and tried to assert that at least one `th` does contain a certain text. However the functions `assertSelectorTextContains`, `assertSelectorTextSame`, and `assertSelectorTextNotContains` will only check the first occurence of the selector, not all occurences. The current example is misleading here. My proposal:

```
Note: the functions `assertSelectorTextContains`, `assertSelectorTextSame`, and `assertSelectorTextNotContains` will only check the first occurence of the selector, not all occurences. If you want to assert that any match of a selector contains some text, use `$this->assertGreaterThan( 0, $crawler->filter('td.specific:contains("some text")')->count());`

### PR DESCRIPTION
**Pull request by javiereguiluz**
Fixes #13036.

### CODE REVIEW

File: testing/functional_tests_assertions.rst [hunk 1]
```diff
@@ -13,16 +13,15 @@ When doing functional tests, sometimes you need to make complex assertions in
...
 
-    $this->assertGreaterThan(
-        0,
-        $crawler->filter('html:contains("Hello World")')->count()
-    );
+    $this->assertSame(301, $client->getResponse()->getStatusCode());
+    $this->assertSame('https://example.com', $client->getResponse()->headers->get('Location'));
 
-Now here is the example with the assertions specific to Symfony::
+This is the same example but using the assertions provided by Symfony::
```
Comment (HeahDude) [inline testing/functional_tests_assertions.rst]:
```diff
-This is the same example but using the assertions provided by
+This is the same example using the assertion provided by
```
?

**Review by HeahDude — APPROVED** [review]

File: testing/functional_tests_assertions.rst [hunk 2]
```diff
@@ -13,16 +13,15 @@ When doing functional tests, sometimes you need to make complex assertions in
...
 
-Here is an example with plain PHPUnit::
+The following example uses plain PHPUnit to assert that the response redirects
+to a certain URL::
 
-    $this->assertGreaterThan(
-        0,
-        $crawler->filter('html:contains("Hello World")')->count()
-    );
+    $this->assertSame(301, $client->getResponse()->getStatusCode());
```
Comment (OskarStark) [inline testing/functional_tests_assertions.rst]: Maybe a bit off topic but I learned that one should use self::assertSame() over $this->assertSame()

Can you give more insights about this @localheinz? Thank you 🙏

File: testing/functional_tests_assertions.rst (hunk 1 above)
Comment (javiereguiluz) [reply testing/functional_tests_assertions.rst]: Done

File: testing/functional_tests_assertions.rst (hunk 2 above)
Comment (javiereguiluz) [reply testing/functional_tests_assertions.rst]: Let's merge this and we can change it later if needed. Thanks.

**Pull request merged**